climada/util/interpolation.py

```diff
@@ -186,3 +202,95 @@ def index_nn_haversine(centroids, coordinates, threshold=THRESHOLD):
+        First column contains latitude, second column contains longitude. Each
+        row is a geographic point
+    threshold : float
+        distance threshold in km over which no neighbor will be found. Those
+        are assigned with a -1 index
+    check_antimedirian: bool
+        if True, recomputes the nn in a strip with lon size equal to threshold
+        around the antimeridian using the Haversine distance.
+
+    Returns
+    -------
+    np.array
+        with so many rows as coordinates containing the centroids indexes
+    """
+    # Construct tree from centroids
+    tree = sp.spatial.KDTree(np.radians(centroids))
+    # Select unique exposures coordinates
+    _, idx, inv = np.unique(coordinates, axis=0, return_index=True,
+                            return_inverse=True)
+
+    # query the k closest points of the n_points using dual tree
+    dist, assigned = tree.query(np.radians(coordinates[idx]), k=1, p=2, workers=-1)
+
+    # Raise a warning if the minimum distance is greater than the
+    # threshold and set an unvalid index -1
+    num_warn = np.sum(dist * EARTH_RADIUS_KM > threshold)
+    if num_warn:
+        LOGGER.warning('Distance to closest centroid is greater than %s'
+                       'km for %s coordinates.', threshold, num_warn)
+        assigned[dist * EARTH_RADIUS_KM > threshold] = -1
+
+    if check_antimeridian:
+        assigned = index_antimeridian_strip(np.radians(centroids),
+                                            np.radians(coordinates[idx]),
+                                            threshold,
+                                            assigned)
+
+    # Copy result to all exposures and return value
+    return np.squeeze(assigned[inv])
+
+def index_antimeridian_strip(centroids, coordinates, threshold, assigned):
+    """Recompute nearest neighbors close to the anti-meridian with the
+    Haversine distance
+
+    Parameters
+    ----------
+    centroids : 2d array
+        First column contains latitude, second column contains longitude.
+        Each row is a geographic point
+    coordinates : 2d array
+        First column contains latitude, second column contains longitude. Each
+        row is a geographic point
+    threshold : float
+        distance threshold in km over which no neighbor will be found. Those
+        are assigned with a -1 index
+    assigned : 1d array
+        coordinates that have assigned so far
+
+    Returns
+    -------
+    np.array
+        with so many rows as coordinates containing the centroids indexes
+
+    """
+
+    coord_strip_bool =  coordinates[:, 1] + np.pi < 1.5*threshold/EARTH_RADIUS_KM
+    coord_strip_bool += coordinates[:, 1] - np.pi >  -1.5*threshold/EARTH_RADIUS_KM
+    if np.any(coord_strip_bool):
+        coord_strip = coordinates[coord_strip_bool]
+        cent_strip_bool = centroids[:, 1] + np.pi < 2.5*threshold/EARTH_RADIUS_KM
+        cent_strip_bool += centroids[:, 1] - np.pi >  -2.5*threshold/EARTH_RADIUS_KM
```

Comment:
```suggestion
        cent_strip_bool |= centroids[:, 1] - np.pi >  -2.5 * threshold / EARTH_RADIUS_KM
```

Reply:
The spaces were remove in a ill attempt to not have lines of more than 80 characters :D

Reply:
In CLIMADA, the convention is to have at most 100 characters per line:
https://github.com/CLIMADA-project/climada_python/blob/30cd7de5363807363bddfe2275cb2eaa248bd1a1/.pylintrc#L218

But yes, you should probably shorten the line by assigning `2.5 * threshold / EARTH_RADIUS_KM` to a variable (like `pad`).